src/utils/checkConnectedToSafe.ts

```diff
@@ -0,0 +1,21 @@
+export async function checkConnectedToSafe(provider: any): Promise<boolean> {
+  if (provider == null) return false
+
+  const wc =
+    (await provider.getWalletConnector?.()) ||
+    (await provider.connection?.getWalletConnector?.()) ||
+    provider.wc ||
+    provider.connection?.wc
+
+  const peerName = wc?.peerMeta?.name
+
+  if (peerName === 'Safe Multisig WalletConnect' || peerName?.startsWith?.('Gnosis Safe')) {
```

Comment:
Should we change this to `startsWith('Safe')` ... just to be on the safe side (no pun intended)

Reply:
Btw the multisig identifies itself as `Gnosis Safe Multisig` so the current logic should work.

see https://github.com/gnosis/safe-react-apps/pull/78/files#diff-791a7865e89890c3f6524eff6f43a645c54438fec18d7064aaff481efa0b9debL27

Reply:
Ah yeah, maybe, though German just merged it with the stricter check, so I guess that's going to stay for now :sweat_smile: